.github/workflows/tests.yml

```diff
@@ -53,8 +53,12 @@ jobs:
           echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
 
       - name: Install poetry
-        run: |
-          curl -fsS https://install.python-poetry.org | python - --preview -y
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+          version: 1.8.5
```

Comment:
Is this change needed? I'm sure it's okay, but it does mean one extra step in the CI supply chain to worry about, so if it's not 100% strictly required, given it's not an official poetry action I'd favour sticking with just "official" curl approach?

Reply:
Since we are using github default runners, simple `pipx install poetry=={version}` is all we need.

Reply:
No problem to revert back to curl. Should I upgrade poetry to 2.0 for the whole project? lock file is not compatible. (that's why I used the action - I could pick the right version easily)

Reply:
For ease of review I'd say keep the poetry upgrade/changes to a separate PR please